### PR TITLE
risk_technical and risk_technical_description invariance fixes #1082

### LINKED ISSUE #1082: The "Technical Risks Description" input field can wrongly be saved with empty spaces
**[Affected versions]:**
- Firefox Release 66.0 and above
- Chrome 73.0.3683.75

**[Affected Platforms]:** 	 	
- All Windows
- All Mac 	 	
- All Linux

**[Prerequisites]:**
- You are logged with your LDAP account on the "https://experimenter.stage.mozaws.net/" website.
- Have an experiment created on the Experimenter stage website.

**[Steps to reproduce]:**
1. Open the browser with the profile from prerequisites and navigate to "https://experimenter.stage.mozaws.net/".
2. Open the created experiment from prerequisites.
3. Click the "Edit" button from "Risks" section.
2. Insert an empty space in the input field.
5. Click the "Save Draft and Finish" button and observe the behavior.

**[Expected result]:**
- The "! This field is required." error message is displayed.

**[Actual result]:**
- The experiment is saved.

**[Notes]:**
- Attached is a screen recording of the issue:
TechDesc

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #1082

### CODE REVIEW

File: app/experimenter/experiments/forms.py [hunk 1]
```diff
@@ -689,6 +689,27 @@ class Meta:
...
+        cleaned_data = super().clean()
+        if (
+            "risk_technical" in cleaned_data
+            and "risk_technical_description" in cleaned_data
+        ):
+            # Both checked, now we need to do an invariance check on these
+            # two. This is to match what's done with jQuery in the form:
+            # the 'risk_technical_description' needs to be required
+            # if 'risk_technical' is truthy.
+            if cleaned_data.get("risk_technical") == "True":
```
Comment (peterbe) [inline app/experimenter/experiments/forms.py]: It's weird to me that these are strings like that. Even after supposed cleaning. I've never seen that before. Is there a better way to do this?

File: app/experimenter/experiments/tests/test_forms.py [hunk 2]
```diff
@@ -1081,6 +1081,46 @@ def test_form_saves_risks(self):
...
+        self.assertFalse(form.is_valid())
+
+        # It would be okey if 'risk_technical' was falsy.
+        data["risk_technical"] = False
+        form = ExperimentRisksForm(
+            request=self.request, data=data, instance=created_experiment
+        )
+        self.assertTrue(form.is_valid())
+
+        # And definitely OK if both at truthy.
```
Comment (jaredkerim) [inline app/experimenter/experiments/tests/test_forms.py]: if both *are truthy

File: app/experimenter/experiments/forms.py (hunk 1 above)
Comment (jaredkerim) [reply app/experimenter/experiments/forms.py]: Yeah that is weird?

File: app/experimenter/experiments/tests/test_forms.py [hunk 3]
```diff
@@ -1081,6 +1081,46 @@ def test_form_saves_risks(self):
         self.assertEqual(experiment.test_builds, data["test_builds"])
         self.assertEqual(experiment.qa_status, data["qa_status"])
 
+    def test_form_risk_technical_invariance(self):
```
Comment (jaredkerim) [inline app/experimenter/experiments/tests/test_forms.py]: I don't think the word 'invariance' is descriptive here, the issue is that if risk_technical is true then risk_technical_description is required, so maybe something like `test_risk_technical_description_optionally_required` would be clearer.

File: app/experimenter/experiments/tests/test_forms.py [hunk 4]
```diff
@@ -1081,6 +1081,46 @@ def test_form_saves_risks(self):
...
+        )
+        self.assertTrue(form.is_valid())
+
+        # And definitely OK if both at truthy.
+        data["risk_technical"] = True
+        data["risk_technical_description"] = "Some text"
+        form = ExperimentRisksForm(
+            request=self.request, data=data, instance=created_experiment
+        )
+        self.assertTrue(form.is_valid())
```
Comment (jaredkerim) [inline app/experimenter/experiments/tests/test_forms.py]: I'd rather have each of these cases tested in separate tests, and also shouldn't we be testing the case where risk_technical is true but risk_technical_description is empty and the form fails to validate?

**Comment by peterbe** [comment]
I'm resigning responsibility to complete this PR. Please feel free to update and do what you want with it.

File: app/experimenter/experiments/forms.py (hunk 1 above)
Comment (robhudson) [reply app/experimenter/experiments/forms.py]: It looks like this is because the base form field is a `ChoiceField` which coerces all values to strings. Since the underlying model for these fields is a boolean we should use `TypedChoiceField` here and provide a proper `coerce` function to make this nicer.

**Comment by robhudson** [comment]
Continuation of this PR in #1148
